src/utils/__tests__/index.test.ts

```diff
@@ -9,6 +11,63 @@ describe('utils', () => {
     })
   })
 
+  describe('resolveOnEvent', () => {
+    const emitter = new EventEmitter()
```

Comment:
I think you should instantiate a new emitter in each test; otherwise the test suite may become nondeterministic.